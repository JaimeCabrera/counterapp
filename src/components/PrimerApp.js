import React from 'react';
import PropTypes from 'prop-types';
const PrimerApp = ({saludo, subtitulo}) => {

   return (
      <>
         <h4>{saludo}</h4>
         <p>{subtitulo}</p>
      </>
   );
}

// propiedades de los componentes
PrimerApp.propTypes = {
   // prop obligatoria .isrequired
   saludo: PropTypes.string.isRequired
}
// valores por defecto de las props
PrimerApp.defaultProps = {
   subtitulo: "soy un subtitulo"
}
export default PrimerApp