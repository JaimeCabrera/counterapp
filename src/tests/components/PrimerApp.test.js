import React from 'react';
import '@testing-library/jest-dom';
import PrimerApp from '../../components/PrimerApp';
import { shallow } from 'enzyme';

describe('Pruebas en <PrimerApp/>', () => {
   // test('debe mostrar el mensaje "Hola soy Ja', () => {
   //    const saludo = "hola soy Ja";
   //    const {getByText} = render(<PrimerApp saludo={saludo} />);
   //    expect(getByText(saludo)).toBeInTheDocument();
   // })
   test('debe mostrar <PrimerApp/> correctamente', () => {
      const saludo = "hola soy Ja";
      const wrapper = shallow(<PrimerApp saludo={saludo} />);
      expect(wrapper).toMatchSnapshot();
   })
   test('debe mostrar el subtitulo enviado por props', () => {
      const saludo = "hola soy Ja";
      const subtitulo = "soy un subtitulo";
      const wrapper = shallow(<PrimerApp saludo={saludo} subtitulo={subtitulo} />);
      const textoParrafo = wrapper.find('p').text();
      expect(textoParrafo).toBe(subtitulo);
   })
   
   
   
})