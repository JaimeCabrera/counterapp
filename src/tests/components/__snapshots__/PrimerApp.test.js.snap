// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimerApp/> debe mostrar <PrimerApp/> correctamente 1`] = `
<Fragment>
  <h4>
    hola soy Ja
  </h4>
  <p>
    soy un subtitulo
  </p>
</Fragment>
`;
